tools/python/migration-scripts/Ingest_S3_to_postgressql/copy_postgres.py

```diff
@@ -0,0 +1,565 @@
+        
+    Returns:
+        bool: True if successful, False otherwise
+    """
+    try:
+            
+        # Get the filename without the path
+        filename = os.path.basename(file_path)
+        
+        # Create the destination path
+        destination = os.path.join(processed_dir, filename)
+        
+        # If the file already exists in the processed directory, add a timestamp
+        if os.path.exists(destination):
+            timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
+            filename_parts = os.path.splitext(filename)
+            new_filename = f"{filename_parts[0]}_{timestamp}{filename_parts[1]}"
+            destination = os.path.join(processed_dir, new_filename)
+            
+        # Move the file
+        shutil.move(file_path, destination)
+        logger.info(f"Moved {file_path} to {destination}")
+        return True
+    except Exception as e:
+            logger.error(f"Failed to move {file_path} to processed directory: {str(e)}")
+            return False
+
+def get_secret_value(secret_name):
+    """
+    Retrieve a secret value from AWS Secrets Manager.
+    
+    Args:
+        secret_name (str): The name or ARN of the secret to retrieve
+        region_name (str): AWS region where the secret is stored (default: 'us-east-1')
+        
+    Returns:
+        dict: The secret value as a dictionary
+        
+    Raises:
+        Exception: If the secret cannot be retrieved
+    """
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(
+        service_name='secretsmanager'
+    )
+    
+    try:
+        # Get the secret value
+        response = client.get_secret_value(
+            SecretId=secret_name
+        )
+    except Exception as e:
+        # Handle exceptions
+        logger.info(f"Error retrieving secret {secret_name}: {str(e)}")
+        raise
+    
+    # Parse and return the secret
+    if 'SecretString' in response:
+        secret = response['SecretString']
+        # Parse JSON string into dictionary
+        return json.loads(secret)
+    else:
+        # If binary, decode it (less common)
+        decoded_binary_secret = base64.b64decode(response['SecretBinary'])
+        return json.loads(decoded_binary_secret)
+
+def validate_sql_identifier(identifier):
+    # Simple validation - alphanumeric and underscore only
+    return bool(re.match(r'^[a-zA-Z0-9_]+$', identifier))
+
+
+if __name__ == '__main__':
+
+    # directory = '/data/csv_files/partition*/*.csv'
+    # database_name = 'postgres_testing'
+    # table_name= 'demolarge_restored'
+
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-d", "--table", help="postgres table name", required=True)
+    parser.add_argument("-t", "--database", help="postgres database name", required=True)
+    parser.add_argument("-s", "--schema", help="Postgres schema where table resides", default='public', required=False)
```

Comment:
I could only get this to work with public schema, and not any others. I was getting error:  - postgres_ingestion - INFO - Last error: relation "tablename" does not exist